### ISSUE TITLE
python no longer compiles with gcc

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#30]
When we converted python to MPI, we at some point made it impossible to compile python and py_wind, etc with regular old gcc.  We need to restore the capability to do this.  Not all machines have mpicc installed, including apparently the linux machines at STScI.  While I expect most serious users will ultimately want to operate in parallel, we ought not to force this.  

We need to provide make files (or a configure file), that allows gcc installation

and we also need to make sure that the code is not written in a way that requires mpi

**Comment by jhmatthews** [comment]
Knox this should now be fixed.

You can compile python with make [D] [CC=gcc] python, and using the following if loop

``` bash
#this is so that the -DMPI_ON flag is only used if you are compiling with mpicc
ifeq (mpicc, $(CC))
    MPI_FLAG = -DMPI_ON
else
    MPI_FLAG =
endif
```

we only use the DMPI_ON flag when compiling with mpicc.

This gives some unused variables flags but works fine. I can probably fixed these with #ifdef loops. Edit: now fixed with commit below. Closing issue.

**Comment by jhmatthews** [comment]
while we're on the subject of compilation, it would be good to fix the 2 errors you get when making python with mpicc:

``` C
get_models.c:157:11: warning: explicitly assigning a variable of type 'int' to itself [-Wself-assign]
          for (m = m; m < NPARS; m++)
               ~ ^ ~
1 warning generated.

thierry.c:255:23: warning: variable 'speed' is uninitialized when used here [-Wuninitialized]
      if (sane_check (speed))
                      ^~~~~
thierry.c:226:18: note: initialize the variable 'speed' to silence this warning
  double r, speed;
                 ^
                  = 0.0
1 warning generated.
```

I wasn't quite sure what these 2 routines were supposed to do so didn't mess with them, but it would be nice to have the compilation message clean so if any warnings come up we know we have introduced them.

**Comment by kslong** [comment]
I have fixed these two errors.  In the second case I simply removed the thierry option from the code as it was never really used.

Knox S. Long
Space Telescope Science Institute
